### PR TITLE
feat(openapi): support dynamic API service icon

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

In `OpenApiDocsView`, this PR makes the icon shown next to the OpenAPI spec title name configurable, rather than being fixed to the `VaultIcon`.

## 🤷 Why

As we onboard new products to the revised `OpenApiDocsView`, we need to be able to show icons other than the Vault icon.

## 🛠️ How

- Adds an optional `serviceProductSlug` option to `OpenApiDocsView` `PageConfig`
    - Falls back to the existing required `productSlug` configuration option
- Uses `ProductIcon` rather than `IconVaultColor16` in `OpenApiOverview`, passing the provided `serviceProductSlug` to drive which icon is shown.

## 📸 Design Screenshots

No design changes, appearance should be identical to upstream.

## 🧪 Testing

- [ ] Visit [/hcp/api-docs/vault-secrets]
    - The icon next to the page title should appear as it does upstream, showing the `Vault` icon
- [ ] Visit the preview tool at [/open-api-docs-preview]
    - When previewing any OpenAPI spec, a generic HashiCorp icon should always be shown
    - Note: we could consider adding a "service product slug" input to the preview tool, but this doesn't feel at all necessary at this point. The preview tool is intended to focus on other aspects of content.

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/1204678746647847/1205543368360332/f
[preview]: https://dev-portal-git-zsopenapi-service-product-icon-hashicorp.vercel.app/
[/hcp/api-docs/vault-secrets]: https://dev-portal-git-zsopenapi-service-product-icon-hashicorp.vercel.app/hcp/api-docs/vault-secrets
[/open-api-docs-preview]: https://dev-portal-git-zsopenapi-service-product-icon-hashicorp.vercel.app/open-api-docs-preview

### CODE REVIEW

File: src/pages/hcp/api-docs/vault-secrets/[[...page]].tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ import type {
  */
 const PAGE_CONFIG: OpenApiDocsPageConfig = {
 	productSlug: 'hcp',
+	serviceProductSlug: 'vault',
```
Comment (zchsh) [inline src/pages/hcp/api-docs/vault-secrets/[[...page]].tsx:28]: We want the `vault` icon for HCP Vault Secrets!

File: src/views/open-api-docs-view/components/open-api-overview/index.tsx [hunk 2]
```diff
@@ -39,14 +39,15 @@ export interface OpenApiOverviewProps {
...
+	serviceProductSlug,
 	statusIndicatorConfig,
 	contentSlot,
 }: OpenApiOverviewProps) {
 	return (
 		<div className={s.overviewWrapper}>
 			<header className={s.header}>
 				<IconTile size="medium" className={s.icon}>
-					<IconVaultColor16 />
+					<ProductIcon productSlug={serviceProductSlug} />
```
Comment (zchsh) [inline src/views/open-api-docs-view/components/open-api-overview/index.tsx:50]: Swaps in `ProductIcon`, previously `IconVaultColor16`

File: src/views/open-api-docs-view/types.ts [hunk 3]
```diff
@@ -173,6 +181,15 @@ export interface OpenApiDocsPageConfig {
...
 	 */
 	productSlug: ProductSlug
+
+	/**
+	 * Optional slug used to theme smaller elements within the service API docs.
+	 * For example, HCP consul will have a `productSlug` of `hcp`,
+	 * but a `serviceProductSlug` of `consul`. If omitted, the `productSlug`
+	 * will be used.
+	 */
+	serviceProductSlug?: ProductSlug
```
Comment (zchsh) [inline src/views/open-api-docs-view/types.ts:191]: Optional at the config level, config will fall back to the required `productSlug`.

File: src/views/open-api-docs-view/types.ts [hunk 4]
```diff
@@ -163,6 +163,14 @@ export interface OpenApiDocsViewProps {
...
 	 */
 	_placeholder: $TSFixMe
+
+	/**
+	 * Product slug to use for the theming of the service itself.
+	 * For example, many product-themed services exist within the broader
+	 * HCP product context. In those cases, the API docs pages would have
+	 * `productData` for `hcp`, and a different `serviceProductSlug` here.
+	 */
+	serviceProductSlug: ProductSlug
```
Comment (zchsh) [inline src/views/open-api-docs-view/types.ts:173]: Required at the view level, we need _some_ `serviceProductSlug` to render.

**Review by kendallstrautman — APPROVED** [review]
Tested with [this spec,](https://github.com/hashicorp/hcp-specs/blob/main/specs/cloud-consul-service/stable/2020-08-26/hcp.swagger.json) looks good!

Screenshot 2023-09-25 at 10 45 24 AM

**Pull request merged**